### DISCUSSION TITLE
Getting percentages of contamination

### DISCUSSION THREAD

**Discussion by sameehan2301** [campbio/celda#373] [Q&A]
Hello, 
Thank you for this nice tool. I was wondering how could I interpret the data? How do I demonstrate that one sample has more ambient RNA contamination as compared to the other? Is there some code that I am missing from the vignette? 

I would like to demonstrate the percentage of contamination in sample A vs sample B.
Please let me know.
Thank you,
Regards,
Sam.

**Comment by joshua-d-campbell** [comment]
Hi @sameehan2301, thanks for using our tool! You can make a boxplot/violin plot to show the differences in contamination between two samples. There are several ways you can do this. For an example, I will import 2 PBMC datasets and run decontX using the [singleCellTK](https://bioconductor.org/packages/release/bioc/html/singleCellTK.html) package:

```
library(singleCellTK)
pbmc3k <- importExampleData("pbmc3k")
pbmc6k <- importExampleData("pbmc6k")
sce <- cbind(pbmc3k, pbmc6k)
sce <- decontX(sce, batch = sce$Sample)
```

If you ran decontX with the data in a *SingleCellExperiment* object, then you can use the [plotSCEViolinColData](https://camplab.net/sctk/v2.5.1/reference/plotSCEViolinColData.html) function from the singleCellTK package:

```
library(singleCellTK)
plotSCEViolinColData(inSCE = sce, coldata = "decontX_contamination", groupBy = "sample", ylab = "Contamination", boxplot = FALSE)
```
<img width="873" alt="image" src="https://user-images.githubusercontent.com/14353256/175430433-0980947e-d7c8-4446-bbb4-b8577df48e38.png">

Or you can extract the contamination vector and use whatever plotting tool you want:

```
contamination <- sce$decontX_contamination
sample <- sce$sample
boxplot(contamination ~ sample)
```

Hope that helps. I'm going to move this to a Discussion so others can see it as well.

**Reply by sameehan2301** [reply]
Thank you for your explanation. I wanted to know how I can get it numerically? 
I am doing the following (starting from a Seurat object) - 

data_1 = as.SingleCellExperiment(data_1)
data_1 = decontX(data_1)
mean(data_1$decontX_contamination)

This gives me a number (between 0-1). Could you please let me know what this number would represent? Could I plot it in a column graph as mean contamination of data_1.? 

Thank you !
Regards,
Sam.

**Comment by joshua-d-campbell** [comment]
Hi @sameehan2301, you are indeed looking at the estimated percentage of contamination for each cell (i.e. `data_1$decontX_contamination` is a vector of cell contamination estimates). Taking the mean of this gives you a sense of the average contamination for your whole dataset. Although I would recommend making a violin plot as sometimes you can get a small number of cells with high contamination that drives the mean up higher. Likewise, you may want to take the median instead of the mean. 

Note that if you want to use the decontaminated count matrix in your downstream Seurat analysis, you can get that matrix our of the SCE object with the following command:

```
decontXcounts(data_1)
```

Additional details of how to work with Seurat objects can be found in our [tutorial](https://www.camplab.net/celda/v1.12.0/articles/articles/decontX_pbmc4k.html).
